test/test_carma_clock.cpp

```diff
@@ -148,6 +190,25 @@ TEST(test_carma_clock, test_sim_time_sleep_until)
     EXPECT_NEAR(SYSTEM_SLEEP_TIME, msCount, 5);
 }
 
+TEST( test_carma_clock, test_sim_time_sleep_for) {
+    CarmaClock clock(true);
+    clock.update(1);
+    
+    auto simTimeNow = clock.nowInMilliseconds();
+    EXPECT_EQ(simTimeNow, 1);
+    auto start = system_clock::now();
+    std::thread t([&clock]() {
+        std::this_thread::sleep_for(std::chrono::milliseconds(SYSTEM_SLEEP_TIME));
+        clock.update(2);
+    });
+    clock.sleep_for(1);
+    auto after = system_clock::now();
+    auto msCount = duration_cast<milliseconds>(after - start).count();
+    t.join();
+    // should have happened after the update occurred in other thread
+    EXPECT_NEAR(SYSTEM_SLEEP_TIME, msCount, 5);
```

Comment:
Shouldn't this be 3 instead of 5? 
```line 195: clock.update(1), wait SYSTEM_SLEEP_TIME,  then clock.update(2);``` The current_time should be 2. Then ```clock.sleep_for(1)```, the clock wait for 1 millisecond on top of current time = 2. so the sleep in total may be 3 milliseconds?

If I can run this unit tests, and get a sense of how it works, it would be much helpful.

Reply:
This 5 is the discrepancy for the unit test. Depending on machine resources this unit test may run faster or slower. The msCount is the difference in machine time for how lone the sleep_for command waits. In this case is should wait for SYSTEM_SLEEP_TIME  since that is how long it takes for the 1ms time step to be updated. We are leaving a 5ms (machine time) +- buffer to account for resource limitations, especially when run in CI/CD environment (Github actions)